histomicstk/FeatureExtraction.py

```diff
@@ -129,6 +144,157 @@ def FeatureExtraction(Label, I, K=128, Fs=6, Delta=8):
     df['Extent'] = Extent
     df['Solidity'] = Solidity
 
-    # df.insert(0, 'Slide', SlideName)
+    FSDGroup = FSDGroup.reshape(Num, Fs)
+
+    df['FSD1'] = FSDGroup[:, 0]
+    df['FSD2'] = FSDGroup[:, 1]
+    df['FSD3'] = FSDGroup[:, 2]
+    df['FSD4'] = FSDGroup[:, 3]
+    df['FSD5'] = FSDGroup[:, 4]
+    df['FSD6'] = FSDGroup[:, 5]
 
     return df
+
+
+def GetBounds(bbox, delta, N):
+    """
+    Returns bounds of object in global label image.
+    """
+    bounds = np.zeros(4, dtype=np.int8)
+    bounds[0] = max(0, math.floor(bbox[0] - delta))
+    bounds[1] = min(N-1, math.ceil(bbox[0] + bbox[2] + delta))
+    bounds[2] = max(0, math.floor(bbox[1] - delta))
+    bounds[3] = min(N-1, math.ceil(bbox[1] + bbox[3] + delta))
+
+    return bounds
+
+
+def InterpolateArcLength(X, Y, L):
+    """
+    Resamples boundary points [X, Y] at L total equal arc-length locations.
+
+    Returns
+    -------
+    iX - L-length vector of horizontal interpolated coordinates with equal
+         arc-length spacing.
+    iY - L-length vector of vertical interpolated coordinates with equal
+         arc-length spacing.
+    """
+
+    # length of X
+    K = len(X)
+
+    # generate spaced points
+    Interval = np.linspace(0, 1, L)
+
+    # get segment lengths
+    Lengths = np.sqrt(
+        np.power(np.diff(X), 2) + np.power(np.diff(Y), 2)
+    )
+
+    # normalize to unit length
+    Lengths = Lengths / Lengths.sum()
+
+    # calculate cumulative length along boundary
+    Cumulative = np.hstack((0., np.cumsum(Lengths)))
+
+    # place points in 'Interval' along boundary
+    Locations = np.digitize(Interval, Cumulative)
+
+    # clip to ends
+    Locations[Locations < 1] = 1
+    Locations[Locations >= K] = K - 1
+    Locations = Locations - 1
+
+    # linear interpolation
+    Lie = np.divide(
+        (Interval - [Cumulative[i] for i in Locations]),
+        [Lengths[i] for i in Locations]
+    )
+
+    tX = np.array([X[i] for i in Locations])
+    tY = np.array([Y[i] for i in Locations])
+    iX = tX + np.multiply(
+        np.array([X[i+1] for i in Locations]) - tX, Lie
+    )
+    iY = tY + np.multiply(
+        np.array([Y[i+1] for i in Locations]) - tY, Lie
+    )
+
+    iXY = collections.namedtuple('iXY', ['iX', 'iY'])
+    Output = iXY(iX, iY)
+
+    return Output
+
+
+def log2space(a, b, n):
+    """
+    Returns n logarithmically-spaced points from 2^a to 2^b.
+    """
+
```

Comment:
I think you can use the [linspace](http://docs.scipy.org/doc/numpy-1.10.0/reference/generated/numpy.linspace.html) function for this -- 
`2**np.linspace(a, b, n, endpoint=True)` or 
`np.power(2, np.linspace(a, b, n, endpoint=True)`